core/app/services/workarea/direct_upload.rb

```diff
@@ -2,17 +2,25 @@ module Workarea
   class DirectUpload
     class InvalidTypeError < RuntimeError; end
 
-    def self.ensure_cors!
-      url = "http#{Rails.application.config.force_ssl ? 's' : ''}://"
-      url += Workarea.config.host
+    def self.ensure_cors!(request = nil)
```

Comment:
Should we just make the url a required arg if it's going to be used everywhere? Then we could clean up that `if` clutter below! Something like: `DirectUpload.ensure_cors!(request.url)`. `request.url` gives me `http://localhost:3000`, which is everything we'd need here. That would also alleviate the need to mock in the test.

Reply:
I'm down with that.